File: translations/codenet/uncommented/C++2J/codellama/Code/s097779718.java
import java.util.Scanner;

public class Euclid {
	public static long gcd(long x, long y) {
		long tmp;

		if (x < y) {
			tmp = x;
			x = y;
			y = tmp;
		}

		if (x % y == 0) return y;
		return gcd(y, x % y);
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		long A = sc.nextLong();
		long B = sc.nextLong();

		System.out.println((A*B) / gcd(A, B));
	}
}

File: translations/codenet/uncommented/C++2Go/granite/Code/s097779718.go

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	const initialCapacity = 4096
	var buf []byte
	scanner.Buffer(buf, initialCapacity)
	scanner.Split(bufio.ScanWords)

	var A, B int64
	scanner.Scan()
	A, _ = strconv.ParseInt(scanner.Text(), 10, 64)
	scanner.Scan()
	B, _ = strconv.ParseInt(scanner.Text(), 10, 64)

	fmt.Println((A * B) / gcd(A, B))
}

func gcd(x, y int64) int64 {
	var tmp int64

	if x < y {
		tmp = x
		x = y
		y = tmp
	}

	for y!= 0 {
		tmp = y
		y = x % y
		x = tmp
	}

	return x
}

